tests/shared/src/test/scala/scala/meta/tests/parsers/TopLevelTermSuite.scala

```diff
@@ -0,0 +1,82 @@
+package scala.meta.tests.parsers
+
+import scala.compat.Platform.EOL
+import scala.meta._
+
+import munit._
+
+class TopLevelTermSuite extends FunSuite {
+  test("allowToplevelTerms simple") {
+    implicit val dialect: Dialect = dialects.Scala3.withAllowToplevelTerms(true)
+
+    val sourceString = """def foo(x: Int): Int = x
+                          |foo(x)
+""".trim.stripMargin
+
+    val tree = sourceString.parse[Source].get
+
+    assert(tree.syntax == sourceString)
+    assert(
+      tree.structure ==
+        """Source(List(
+          |Defn.Def(Nil, Term.Name("foo"), Nil, List(List(Term.Param(Nil, Term.Name("x"), Some(Type.Name("Int")), None))), Some(Type.Name("Int")), Term.Name("x")), 
+          |Term.Apply(Term.Name("foo"), List(Term.Name("x")))))
+          |""".trim.stripMargin.split('\n').mkString("")
+    )
+  }
+
+  test("allowToplevelTerms with package crashes") {
+    implicit val dialect: Dialect = dialects.Scala3.withAllowToplevelTerms(true)
+
+    val sourceString = """package bar
+                          |def foo(x: Int): Int = x
+                          |foo(x)
+""".trim.stripMargin
+
+    val e = intercept[ParseException] {
+      sourceString.parse[Source].get
+    }
+    assert(e.getMessage().contains("illegal start of definition package"))
+  }
+
+  test("allowToplevelTerms and allowPackageStatementsWithToplevelTerms with package") {
+    implicit val dialect: Dialect =
+      dialects.Scala3.withAllowToplevelTerms(true).withAllowPackageStatementsWithToplevelTerms(true)
+
+    val sourceString = """package bar
+                         |def foo(x: Int): Int = x
+                         |foo(x)
+""".trim.stripMargin
+
+    val tree = sourceString.parse[Source].get
+
+    assert(tree.syntax == sourceString)
+    val expected =
+      """Source(List(Pkg(Term.Name("bar"), List(
+                     |Defn.Def(Nil, Term.Name("foo"), Nil, List(List(Term.Param(Nil, Term.Name("x"), Some(Type.Name("Int")), None))), Some(Type.Name("Int")), Term.Name("x")), 
+                     |Term.Apply(Term.Name("foo"), List(Term.Name("x"))))
+                     |)))""".trim.stripMargin.split('\n').mkString("")
+    assert(tree.structure == expected)
```

Comment:
Let's use assertEquals, which  should give us a better diff in case of issues.

Reply:
Done.